### PR TITLE
feat: Adding minio with dex auth to nerc-ocp-obs cluster

### PR DESCRIPTION
**Pull request by dheerajodha**
Adding support for object stores in the nerc-ocp-obs cluster with authentication

### CODE REVIEW

**Comment by computate** [comment]
@dheerajodha Looking good! Let's include a PVC patch like this for `20Ti` storage. 
https://github.com/OCP-on-NERC/nerc-ocp-config/blob/main/minio/overlays/nerc-ocp-test/persistentvolumeclaims/patch-pvc.yaml

**Comment by dheerajodha** [comment]
Hey @computate, I've added the PVC patch in my latest commit, PTAL.

**Comment by computate** [comment]
Excellent @dheerajodha , one more change we should add in the [obs cluster dex ConfigMap here](https://github.com/OCP-on-NERC/nerc-ocp-config/blob/main/dex/overlays/nerc-ocp-obs/configmaps/files/config.yaml). Can you please add these 2 additional `staticClients`?  I have added the corresponding `MINIO_IDENTITY_OPENID_CLIENT_SECRET` and `AI_TELEMETRY_AUTH_SECRET` to vault, so that they will be added to the existing ExternalSecret. 

```yaml
  - id: minio
    name: MinIO
    redirectURIs:
      - https://minio-console-minio.apps.obs.nerc.mghpcc.org/oauth_callback
      - https://minio-minio.apps.obs.nerc.mghpcc.org/oauth_callback
    secretEnv: MINIO_IDENTITY_OPENID_CLIENT_SECRET
  - id: ai-telemetry
    name: AI Telemetry
    redirectURIs:
      - https://keycloak.apps.obs.nerc.mghpcc.org/realms/NERC/broker/OpenShift/endpoint
    secretEnv: AI_TELEMETRY_AUTH_SECRET
```

**Comment by computate** [comment]
I have also added a `nerc-ocp-obs/minio/minio-admin-credentials` secret in vault containing the `MINIO_IDENTITY_OPENID_CLIENT_SECRET`, `MINIO_ROOT_PASSWORD`, and `MINIO_ROOT_USER`, which the minio deployment will read to connect to dex, and provide an admin user for minio.

**Comment by dheerajodha** [comment]
Thank you @computate for the explanation, this makes sense! I've added the 2 staticClients. PTAL

File: minio/overlays/nerc-ocp-obs/files/minio-config.env [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Documentation: https://min.io/docs/minio/linux/reference/minio-server/settings/iam/openid.html
+
+MINIO_IDENTITY_OPENID_CONFIG_URL=https://dex-dex.apps.ocp-obs.nerc.mghpcc.org/.well-known/openid-configuration
```
Comment (computate) [inline minio/overlays/nerc-ocp-obs/files/minio-config.env]: One more thing @dheerajodha , we need to change `ocp-obs` to `obs` here.

**Review by computate — APPROVED** [review]

File: minio/overlays/nerc-ocp-obs/files/minio-config.env (hunk 1 above)
Comment (dheerajodha) [reply minio/overlays/nerc-ocp-obs/files/minio-config.env]: Thanks! updated.

**Review by computate — APPROVED** [review]
Great work with all these fixes!

**Review by larsks — APPROVED** [review]

**Review by tssala23 — APPROVED** [review]

**Pull request merged**